pkg/util/source/inspector.go

```diff
@@ -336,6 +336,11 @@ func (i *baseInspector) discoverKamelets(meta *Metadata) {
 }
 
 func (i *baseInspector) addDependencies(uri string, meta *Metadata, consumer bool) error {
+	if !strings.Contains(uri, ":") {
```

Comment:
As Pasquale already pointed out, it appeares this check should be smarter. Example:
https://github.com/apache/camel-kamelets/blob/main/kamelets/elasticsearch-search-source.kamelet.yaml#L110

My suggestion would be, instead of checking the uri at the beginning of this func, what about examining the `scheme` and `candidateComp` values inside the block at line 345 after calling `DecodeComponent()` method?